### PR TITLE
Reinitialize Active Storage

### PR DESCRIPTION
**Pull request by brianp**
### Summary

Active Storage was setup during a previous upgrade but never used. This PR brings it back to life.

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

- [] ~Added a CHANGELOG entry~

### CODE REVIEW

File: db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# This migration comes from active_storage (originally 20190112182829)
+class AddServiceNameToActiveStorageBlobs < ActiveRecord::Migration[6.0]
+  def up
+    unless column_exists?(:active_storage_blobs, :service_name)
```
Comment (sean-yeoh) [inline db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb:4]: What's the `service_name` column for?

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -90,6 +90,7 @@
     ensure
       DatabaseCleaner.clean_with(:truncation)
       Attachment.all.each(&:delete)
+      FileUtils.rm_rf(Rails.root.join("tmp/storage"))
```
Comment (sean-yeoh) [inline spec/rails_helper.rb:93]:
```suggestion
      FileUtils.rm_rf(Rails.root.join('tmp/storage'))
```

File: db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb (hunk 1 above)
Comment (brianp) [reply db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb:4]: If we were using AWS, or some other third party file storage I believe the name of the service used to store that particular blob will be added to that column. Not sure what ours will contain. Maybe blank, maybe "filestore".

Comment (sean-yeoh) [reply db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb:4]: Ahh okay, move along then.

File: db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      add_column :active_storage_blobs, :service_name, :string
+
+      if configured_service = ActiveStorage::Blob.service.name
+        ActiveStorage::Blob.unscoped.update_all(service_name: configured_service)
+      end
```
Comment (etdsoft) [inline db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb:9]: this seems like a weird migration:

1. create a brand new column
2. update all recrods with a new default
3. update column definition to disallow null columns

Can't 1-2 be combined into a single `add_column` with `default: configured_service` call? It seems that we're defining a service for all ./config/environments/ possible.

Or perhaps you can combine 2-3 with a call to [change_column_null](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-change_column_null) providing a 4th argument of `configured_service`)

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -90,6 +90,7 @@
     ensure
       DatabaseCleaner.clean_with(:truncation)
       Attachment.all.each(&:delete)
```
Comment (etdsoft) [inline spec/rails_helper.rb:92]: I'm wondering if we could save some time with a `.rm_rf` call on `Attachment.pwd` here?

File: db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb (hunk 1 above)
Comment (etdsoft) [reply db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb:4]: Why do you need a guard against the column existing?

**Review by etdsoft — APPROVED** [review]

File: db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb (hunk 3 above)
Comment (brianp) [reply db/migrate/20210316202859_add_service_name_to_active_storage_blobs.active_storage.rb:9]: Same here. These migrations were generated by the rails upgrade command so I never questioned the contents.

File: spec/rails_helper.rb (hunk 4 above)
Comment (brianp) [reply spec/rails_helper.rb:92]: Seems like a good recommendation for an aside PR.

**Pull request merged**